src/effect/looper.rs

```diff
@@ -0,0 +1,63 @@
+use crate::{
+    audio::midi,
+    audio_unit::{self, looper::Message, AudioUnit},
+    config::LooperConfig,
+    effect::Effect,
+    Result,
+};
+use cpal::StreamConfig;
+use std::sync::mpsc::Sender;
+use wmidi::MidiMessage;
+
+pub struct Looper {
+    config: LooperConfig,
+    split: audio_unit::Split,
+    messages: Sender<Message>,
+}
+
+impl Looper {
+    pub fn new(config: LooperConfig, stream_config: &StreamConfig) -> Result<Self> {
+        let (looper, messages) = audio_unit::Looper::new(stream_config, config.max_ms);
+        let transparent = audio_unit::Transparent::new();
+
+        let split = audio_unit::Split::new(vec![looper.boxed(), transparent.boxed()])?;
```

Comment:
Like you mentioned before, it's kind of weird that every effect has to do this, but I didn't want to deal with that in this PR.

Reply:
And likely lots of types of effects wouldn't actually pass through the clean signal (whether they have a "wet/dry" concept or just always mess with the incoming signal)

Reply:
Hmm, in that case does it actually make sense to leave it up the effect then? Maybe with a helper to remove some of the boilerplate?

Reply:
> Hmm, in that case does it actually make sense to leave it up the effect then? Maybe with a helper to remove some of the boilerplate?

Ya that sounds like the right approach ("opt-in")